jenkins_pipelines/scripts/maintenance_json_generator.py

```diff
@@ -148,174 +152,160 @@
+    "50": nodesdict50
+}
 
-def parse_args():
+def parse_args() -> argparse.Namespace:
     parser = argparse.ArgumentParser(
-        description="This script reads the open qam-manager requests and creates a json file that can be fed in the "
-                    "BV testsuite pipeline")
+        description="This script reads the open qam-manager requests and creates a json file that can be fed to the BV testsuite pipeline"
+    )
     parser.add_argument("-v", "--version", dest="version",
-                        help="Version of SUMA you want to run this script for, the options are 43 for 4.3 and 50 for "
-                             "5.0. The default is 43 for now",
-                        default="43", action='store')
+        help="Version of SUMA you want to run this script for, the options are 43 for 4.3 and 50 for 5.0. The default is 43 for now",
+        choices=["43", "50"], default="43", action='store', 
+    )
     parser.add_argument("-i", "--mi_ids", dest="mi_ids", help="MI IDs", default=None, action='store')
+    parser.add_argument("-e", "--no_embargo", dest="embargo_check", help="Reject MIs under embargo",  action='store_true')
 
-    args = parser.parse_args()
+    args: argparse.Namespace = parser.parse_args()
     return args
 
-
-def read_requests():
-    # Find open requests
-    result = object
+def osc_ibs_request(cmd: str) -> str:
+    result: subprocess.CompletedProcess[bytes] = None
     try:
-        # TODO Find a better way to query the open requests, this is fragile because it depends on external utils
-        #  being there.
-        result = subprocess.run(["osc --apiurl https://api.suse.de qam open -G qam-manager"], shell=True,
-                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        result = subprocess.run([f"osc --apiurl https://api.suse.de {cmd}"], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     except subprocess.CalledProcessError:
         print("The ibs command failed for some reason")
-    output = result.stdout.decode('utf-8')
-    lines = output.splitlines()
-    # Create empty list to add the maintenance incidents from the output
-    mi_ids = []
-    for line in lines:
-        if "ReviewRequest" in line:
-            line1 = line.rstrip()
-            line2 = line1.split(sep=":")
-            mi_id = line2[3]
-            mi_ids.append(mi_id)
+
+    output: str = result.stdout.decode('utf-8')
+    return output
+    
+def read_requests() -> set[str]:
+    # Find open requests
+    # TODO Find a better way to query the open requests, this is fragile because it depends on external utils
+    # being there.
+    output: str = osc_ibs_request("qam open -G qam-manager")
+    lines: list[str] = output.splitlines()
+    # Create a set of the maintenance incidents from the output
+    mi_ids = { line.rstrip().split(sep=":")[3] for line in lines if "ReviewRequest" in line }
     return mi_ids
 
+@cache
+def create_url(mi_id:str, suffix: str) -> str:
```

Comment:
Code should either have all the returns or no returns at all, thus:
```suggestion
def create_url(mi_id:str, suffix: str) -> str | None:
```

Reply:
Good catch.
I'd probably return an empty string instead of `None` though.
Not that it matters that much for the context of this script , as they both would evaluate to False where the return is used.